### ISSUE TITLE
Timeout exception is not catched

### ISSUE THREAD

**Issue by mhbacks** [nextcloud/android#5437]
### Steps to reproduce
1. do something on a slow server

### Expected behaviour
- an toast should be displayed, but please add a settings option to set the waitingtime

### Actual behaviour
- an exception is raised and the app crashed

### Environment data
Android version:

Device model: 

Stock or customized system:

Nextcloud app version:
3.10.1

Nextcloud server version:

#### Android Exception message 
```
************ CAUSE OF ERROR ************

java.net.SocketTimeoutException: timeout
	at okio.SocketAsyncTimeout.newTimeoutException(Okio.kt:149)
	at okio.AsyncTimeout.access$newTimeoutException(AsyncTimeout.kt:162)
	at okio.AsyncTimeout$source$1.read(AsyncTimeout.kt:340)
	at okio.RealBufferedSource.indexOf(RealBufferedSource.kt:449)
	at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.kt:333)
	at okhttp3.internal.http1.Http1Codec.readHeaderLine(Http1Codec.java:215)
	at okhttp3.internal.http1.Http1Codec.readResponseHeaders(Http1Codec.java:189)
	at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:88)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:127)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:257)
	at okhttp3.RealCall.execute(RealCall.java:93)
	at com.nextcloud.common.OkHttpMethodBase.execute(OkHttpMethodBase.kt:126)
	at com.nextcloud.common.NextcloudClient.execute(NextcloudClient.kt:84)
	at com.owncloud.android.lib.resources.activities.GetActivitiesRemoteOperation.run(GetActivitiesRemoteOperation.java:131)
	at com.nextcloud.common.NextcloudClient.execute(NextcloudClient.kt:80)
	at com.owncloud.android.ui.fragment.FileDetailActivitiesFragment.lambda$fetchAndSetData$5$FileDetailActivitiesFragment(FileDetailActivitiesFragment.java:331)
	at com.owncloud.android.ui.fragment.-$$Lambda$FileDetailActivitiesFragment$N7POe6TZvoGcuV2qP-niPxY_7t4.run(Unknown Source:12)
	at java.lang.Thread.run(Thread.java:764)
Caused by: java.net.SocketException: socket is closed
	at com.google.android.gms.org.conscrypt.ConscryptFileDescriptorSocket$SSLInputStream.read(:com.google.android.gms@20104028@20.1.04 (100400-288960190):8)
	at okio.InputStreamSource.read(Okio.kt:93)
	at okio.AsyncTimeout$source$1.read(AsyncTimeout.kt:129)
	... 26 more

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30100190
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: OnePlus
Device: OnePlus6
Model: ONEPLUS A6003
Id:
Product: OnePlus6

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: 1909112330

```

**Comment by ezaquarii** [comment]
This is fixed in https://github.com/nextcloud/android-library/pull/394

**Comment by tobiasKaminsky** [comment]
Can you try out our dev version (you can install it in parallel) and give us feedback:
https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#dev-release

**Comment by AndyScherzinger** [comment]
Duplicate of #5331
